testimony/__init__.py

```diff
@@ -450,6 +462,43 @@ def bugs_report(testcases):
     )
 
 
+def tags_report(testcases):
+    """Lists the test cases matching the input tag."""
+    result = {}
+    result['tagged_testcases'] = []
+    if not SETTINGS['input_tags']:
+        print 'Input tags required for this report.  See testimony --help'
+        sys.exit()
+    # Change the input tags to lower case
+    input_tags_list = [tag.lower() for tag in SETTINGS['input_tags']]
+    for _, tests in testcases.items():
+        for testcase in tests:
+            if testcase.tags:
+                # Split the test case tags with ',' and change to lowercase
+                # and trim empty spaces
+                testcase_tags_list = [
+                    tag.lower().strip()
+                    for tag in testcase.tags.split(',')
+                ]
+                # Check if any items in either list match. If so, derive the
+                # full path of the test case. Expected sample output:
+                # `tests.test_sample.Testsample1.test_positive_login_1`
+                if (len(set(testcase_tags_list).intersection(
```

Comment:
The following will make code easier to read (considering that testcase_tags_list was generated by a set comprehension):

```python
if testcase_tags_list.intersection(input_tags_list):